plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -756,6 +757,14 @@ private void registerPrebuiltModel(MLRegisterModelInput registerModelInput, MLTa
             throw new IllegalArgumentException("This model is not in the pre-trained model list, please check your parameters.");
         }
         modelHelper.downloadPrebuiltModelConfig(taskId, registerModelInput, ActionListener.wrap(mlRegisterModelInput -> {
+            mlTask.setFunctionName(mlRegisterModelInput.getFunctionName());
+            mlTaskManager
+                .updateMLTask(
```

Comment:
Can you add testing for this? Code coverage is dropping.

Reply:
I have added one. Please check it.